### PR TITLE
#Feature-ImplementFriendsAccess

### PR DESCRIPTION
**Pull request by bod9i**

### CODE REVIEW

File: GhostNetwork.Gateway.Api/Users/SecuritySettingsController.cs [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            this.currentUserProvider = currentUserProvider;
+        }
+
+        [HttpGet]
+        [ProducesResponseType(StatusCodes.Status200OK)]
+        [ProducesResponseType(StatusCodes.Status404NotFound)]
+        public async Task<IActionResult> GetUserSecuritySettings()
+        {
+            var settings = await usersStorage.SecuritySettings.FindByProfileAsync(new Guid(currentUserProvider.UserId));
+            if (settings == null)
```
[the remark this answers is not in the document]
Comment (Boberneprotiv) [reply GhostNetwork.Gateway.Api/Users/SecuritySettingsController.cs:29]: don't forget to remove ProducesResponseType attribute

**Review by volodymyr-borodin — CHANGES_REQUESTED** [review]
PR doesn't passed auto tests

File: GhostNetwork.Gateway/GhostNetwork.Gateway.csproj [hunk 2]
```diff
@@ -7,6 +7,7 @@
 
     <ItemGroup>
         <PackageReference Include="Boberneprotiv.Domain" Version="0.0.3" />
+        <PackageReference Include="GhostNetwork.Profiles" Version="1.3.5" />
```
Comment (volodymyr-borodin) [inline GhostNetwork.Gateway/GhostNetwork.Gateway.csproj:10]: infrastructure already has this package reference

File: GhostNetwork.Gateway/Users/SecuritySection/SecuritySettingModel.cs [hunk 3]
```diff
@@ -0,0 +1,53 @@
...
+            return new SecuritySettingModel(
+                userId,
+                new SecuritySettingSection(AccessLevel.Everyone, Enumerable.Empty<Guid>()),
+                new SecuritySettingSection(AccessLevel.Everyone, Enumerable.Empty<Guid>()),
+                new SecuritySettingSection(AccessLevel.Everyone, Enumerable.Empty<Guid>()),
+                new SecuritySettingSection(AccessLevel.Everyone, Enumerable.Empty<Guid>()),
+                new SecuritySettingSection(AccessLevel.Everyone, Enumerable.Empty<Guid>()));
+        }
+
+        public void Update(SecuritySettingSection posts, SecuritySettingSection friends)
```
Comment (volodymyr-borodin) [inline GhostNetwork.Gateway/Users/SecuritySection/SecuritySettingModel.cs:47]: do you need this methid?

File: GhostNetwork.Gateway/Users/SecuritySection/SecuritySettingModel.cs [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+            Posts = posts;
+            Friends = friends;
+            Followers = followers;
+            Comments = comments;
+            ProfilePhoto = profilePhoto;
+        }
+
+        public Guid UserId { get; }
+
+        public SecuritySettingSection Posts { get; private set; }
```
Comment (volodymyr-borodin) [inline GhostNetwork.Gateway/Users/SecuritySection/SecuritySettingModel.cs:26]: please remove sections your are not going to use in this PR

File: GhostNetwork.Gateway/Users/IUsersRelationsStorage.cs [hunk 5]
```diff
@@ -11,6 +11,8 @@ public interface IUsersRelationsStorage
 
         Task<IEnumerable<UserInfo>> GetFollowersAsync(Guid user, int take, int skip);
 
+        Task<bool> IsFriendAsync(Guid userId, Guid ofUserId);
```
Comment (volodymyr-borodin) [inline GhostNetwork.Gateway/Users/IUsersRelationsStorage.cs:14]: do you still need this method?

File: GhostNetwork.Gateway/Users/ISecuritySettingsResolver.cs [hunk 6]
```diff
@@ -0,0 +1,12 @@
...
+using System;
+using System.Threading.Tasks;
+
+namespace GhostNetwork.Gateway.SecuritySettings;
+
+public interface ISecuritySettingsResolver
+{
+    Task<DomainResult> ResolveAccessAsync(Guid userId);
+
+    Task<DomainResult> ResolveAccessAsync(string userId);
```
Comment (volodymyr-borodin) [inline GhostNetwork.Gateway/Users/ISecuritySettingsResolver.cs:11]: this test used only in tests. i this it’s better to remove it

File: GhostNetwork.Gateway/Users/ISecuritySettingsResolver.cs [hunk 7]
```diff
@@ -0,0 +1,12 @@
+﻿using Domain;
+using System;
+using System.Threading.Tasks;
+
+namespace GhostNetwork.Gateway.SecuritySettings;
+
+public interface ISecuritySettingsResolver
```
Comment (volodymyr-borodin) [inline GhostNetwork.Gateway/Users/ISecuritySettingsResolver.cs:7]: what’s the reason for this interface?

**Review by volodymyr-borodin — CHANGES_REQUESTED** [review]
please double check for unused code. there are some methods without links

File: GhostNetwork.Gateway/GhostNetwork.Gateway.csproj (hunk 2 above)
Comment (bod9i) [reply GhostNetwork.Gateway/GhostNetwork.Gateway.csproj:10]: Yes but how I can use models from profile-service in interface that store on "Domain" project?

File: GhostNetwork.Gateway/Users/ISecuritySettingsResolver.cs (hunk 7 above)
Comment (bod9i) [reply GhostNetwork.Gateway/Users/ISecuritySettingsResolver.cs:7]: Describing of functionality for any resolvers. Remove it and write resolvers without any interface?

Comment (volodymyr-borodin) [reply GhostNetwork.Gateway/Users/ISecuritySettingsResolver.cs:7]: Yes, remove it. You don’t use this interface at all

File: GhostNetwork.Gateway/GhostNetwork.Gateway.csproj (hunk 2 above)
Comment (volodymyr-borodin) [reply GhostNetwork.Gateway/GhostNetwork.Gateway.csproj:10]: profile service models should be used only in infrastructure layer

**Comment by volodymyr-borodin** [comment]
Please double check your code, there are still some unused parts

**Review by volodymyr-borodin — APPROVED** [review]

**Review by volodymyr-borodin — APPROVED** [review]

**Pull request merged**